internal/retryhttp/client.go

```diff
@@ -37,6 +41,44 @@ var (
 	DefaultTimeout = time.Second * 30
 	DefaultRetries = 5
 	DefaultClient  = NewClient(DefaultTimeout, DefaultRetries)
+
+	// A regular expression to match the error returned by net/http when the
+	// configured number of redirects is exhausted. This error isn't typed
+	// specifically so we resort to matching on the error string.
+	redirectsErrorRe = regexp.MustCompile(`stopped after \d+ redirects\z`)
+
+	// A regular expression to match the error returned by net/http when the
+	// scheme specified in the URL is invalid. This error isn't typed
+	// specifically so we resort to matching on the error string.
+	schemeErrorRe = regexp.MustCompile(`unsupported protocol scheme`)
```

Comment:
Aren't these system errors? Meaning they would be localized and these regexes would only match on English systems.

Reply:
I should have noted that these regexs are also from the retry library that we are using. I grabbed them as the function I added to check retries is modifying the behavior of the default one. 

These string values are coming from the `net/http` package which I _believe_ isn't localized but also since we are statically building our binary I don't think they would be presented in different languages.

https://cs.opensource.google/go/go/+/master:src/net/http/client.go;l=812?q=%22stopped%20after%22&ss=go%2Fgo:src%2Fnet%2Fhttp%2F

https://cs.opensource.google/go/go/+/master:src/net/http/transport.go;l=560?q=%22unsupported%20protocol%20scheme%22&ss=go%2Fgo:src%2Fnet%2Fhttp%2F